core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -330,6 +347,7 @@ private[spark] class SparkSubmit extends Logging {
       }
     }
 
+    // Fill in all spark properties of SparkConf
```

Comment:
I underline this here because it should be a few lines above but not sure if it matters. The issue is a few lines above I have to use `args.sparkProperties` instead. The other problem is that at line 802 we re-fill the properties if missing... not sure why.

Reply:
I agree we should clean it up.